app/models/comment.rb

```diff
@@ -0,0 +1,9 @@
+class Comment < ApplicationRecord
+  belongs_to :post
+  belongs_to :author
+  validates :comment_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
+  def increment_comment_counter
+    post.increment!(:comment_counter)
+  end
```

Comment:
- [ Required ] I think it would be better if you use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#increment_comment_counter` method every time a new comment has been saved to database, then make the `increment_comment_counter` method private. Please see the example below and rectify this.

## Example code 
```ruby
  after_save :increment_comment_counter

  private

  def increment_comment_counter
    post.increment!(:comment_counter)
  end
```

Reply:
@Felix45 , Thanks a lot for this review. I'm being genuine.